core/services/job/orm.go

```diff
@@ -412,11 +413,22 @@ func (o *orm) InsertWebhookSpec(webhookSpec *WebhookSpec, qopts ...pg.QOpt) erro
 
 func (o *orm) InsertJob(job *Job, qopts ...pg.QOpt) error {
 	q := o.q.WithOpts(qopts...)
-	query := `INSERT INTO jobs (pipeline_spec_id, name, schema_version, type, max_task_duration, ocr_oracle_spec_id, ocr2_oracle_spec_id, direct_request_spec_id, flux_monitor_spec_id,
+	var query string
+
+	// if job has id, emplace otherwise insert with a new id.
+	if job.ID == 0 {
+		query = `INSERT INTO jobs (pipeline_spec_id, name, schema_version, type, max_task_duration, ocr_oracle_spec_id, ocr2_oracle_spec_id, direct_request_spec_id, flux_monitor_spec_id,
```

Comment:
An alternative approach might be to use a `null.Int64` and translate a `0` jobId into a null value, then you can get away with only one insert (I think)